### DISCUSSION TITLE
Does vite-plugin have a dependency on Babel?

### DISCUSSION THREAD

**Discussion by AkimaLunar** [microsoft/griffel#523] [Q&A]
I am working on a project that uses SWC instead of Babel and seeing errors like the one below after adding `griffel()` to the plugins array, as per docs [here](https://griffel.js.org/react/ahead-of-time-compilation/with-vite). The errors I am seeing suggest that I need to configure Babel, which I don’t have for this project.

```
Add @babel/preset-flow (https://github.com/babel/babel/tree/main/packages/babel-preset-flow) to the 'presets' section of your Babel config to enable transformation.
If you want to leave it as-is, add @babel/plugin-syntax-flow (https://github.com/babel/babel/tree/main/packages/babel-plugin-syntax-flow) to the 'plugins' section to enable parsing. (x4)
3:29:46 PM [vite] Internal server error: /app/src/ui/theme/index.ts: Support for the experimental syntax 'flow' isn't currently enabled (1:8):

> 1 | export type { Theme} from './theme';
    |        ^
```

Adding `.babelrc` freezes the dev server with errors like: `[BABEL] Note: The code generator has deoptimised the styling of .../app/node_modules/.vite/deps/chunk-KDWZNFJ2.js as it exceeds the max of 500KB.` and excluding `node_modules` doesn’t fix it.

Either way, ideally Babel doesn’t transpile files at all in this project, because it already has SWC.

_Any ideas on how to set up Griffel with this project?_

**Comment by layershifter** [comment]
@AkimaLunar yes, Babel is used in `@wyw-in-js/*` packages under hood. Are you able to provide a simple repro case as GH repo? I will take a look then.

**Reply by layershifter** [reply]
To clarify, these is a condition that adds support for TypeScript: https://github.com/Anber/wyw-in-js/blob/7cc243bb5c083d1fff0a44c5e0597374289d0684/packages/transform/src/shaker.ts#L50-L71

But it looks that something does not work...
